openebs/openebs
# [TA1787, TA1789] Rewrite PromQl to get the cumulative stats and recreate Dashboard

On branch US2355-dashboard
Changes to be committed:
	modified:   openebs-monitoring-pg.yaml
	modified:   openebs-pg-dashboard.json

1. Why is this change necessary?
-  PR openebs/maya#379 involves lot of change in logic w.r.t stats calculation
   so existing dashboard will not work.
-  To incorporate those changes in the dashboard.

2. How does it address the issue?
-  This commit make changes in the PromQL and dashboard JSON to get the
   cumulative stats.

3. What side effects does this change have?
-  Docker images with tags < 0.6.0 will not be compatible with this dashboard.

4. How to verify this change?
-  Make sure you are using the m-exporter's image tag > `0.6.0`
-  Deploy the openebs-monitoring-pg.yaml on kubernetes cluster.
-  Import the new dashboard and it should work.

5. Any specific message for reviewer ?
-  Added namespace field in the Prometheus's template.
-  Replaced the job name from `openebs-prometheus-server` to `prometheus` only.
   Prometheus's dashboard is breaking with the current specs, because grafana is autoconfigured with
   job name `prometheus`.

Signed-off-by: Utkarsh Mani Tripathi <utkarshmani1997@gmail.com>
